### ISSUE TITLE
Prisma Plugin requires nullable fields for required files

### ISSUE THREAD

**Issue by LawJolla** [hayes/pothos#1013]
Hi Hayes and anyone looking in...

I'm trying to upgrade from Prisma 4.10 to 5.2.  However, Prisma Plugin appears to require that I make non-null fields nullable.  The following video also happens with `tsc -b`

My dependencies..
```
...
 "dependencies": {
    "@pothos/core": "^3.33.1",
    "@pothos/plugin-prisma": "^3.59.0",
    "@pothos/plugin-prisma-utils": "0.11.0",
    "@prisma/client": "5.2.0",
    "prisma": "5.2.0"
    ....
  },
...
```

tsconfig is in strict mode.

https://github.com/hayes/pothos/assets/17485128/d1b33173-7d71-458b-a8fd-6fc947d3775d

Please let me know how I can help.  I'm at a dead end at the moment.

Thanks for any help and the amazing work!

**Comment by hayes** [comment]
Can you share your full teconfig?

**Comment by LawJolla** [comment]
... I feel like when I go to the dentist and haven't flossed enough. 🤣

```
{
    "compilerOptions": {
        "baseUrl": "./",
        "composite": false,
        "declaration": false,
        "declarationMap": false,
        "esModuleInterop": true,
        "forceConsistentCasingInFileNames": true,
        "inlineSources": false,
        "isolatedModules": true,
        "moduleResolution": "nodenext",
        "noUnusedLocals": false,
        "noUnusedParameters": false,
        "preserveWatchOutput": true,
        "skipLibCheck": true,
        "strict": true,
        "useUnknownInCatchVariables": false,
        "jsx": "react-jsx",
        "paths": {
            "@/*": [
                "./src/*"
            ]
        },
        "lib": [
            "esnext",
            "dom"
        ],
        "module": "nodenext",
        "target": "esnext",
        "allowSyntheticDefaultImports": true,
        "outDir": "./dist",
        "rootDir": "./src",
        "noEmit": true,
        "strictNullChecks": true
    },
     "include": [
        "src"
    ],
    "exclude": [
        "node_modules"
    ]
}
```

Thank you!

**Comment by LawJolla** [comment]
Quick video showing the failure on Prisma 5.2, builds on 4.10.  (Sorry for the slowness.. on the computer at the office)


https://github.com/hayes/pothos/assets/17485128/5083d92d-67b1-418a-b62c-2597a8189596

**Comment by hayes** [comment]
Thanks for sharing those videos.  Unfortunately I haven't been able to reproduce anything like this.   If you want to try to create a reproduction, I have this repo: https://github.com/hayes/pothos-prisma-alias-bug. that I have been using to reproduce some other issues.  I added your tsconfig, but haven't been able to reproduce the issue yet.

Based on your video, it looks like its specific to the expose methods, but I can't tell much else.

Hovering over the t.expose method might show more details about what generics are being inferred, although not sure if that will help or not.

Some things that might help:

* typescript version
* the prisma schema (or just the config you have for generators, and any preview features you have enabled, maybe what kind of database you are using)
* does this happen on every type, and every field?  Is it only expose, or do other methods also run into this issue


Unfortunately I don't have too much to go off of right now

**Comment by LawJolla** [comment]
Thanks Hayes!

I've been working using your repo and mine to isolate the issue / reproduce it.  I'm close.  In the interim, I did manage to make a scaled down version error on dev with the following error.  Maybe this helps?

Either way I'll keep trying to make a reproduction.
![nodemon json — wk-mono-repo 2023-09-06 at 4 21 59 PM](https://github.com/hayes/pothos/assets/17485128/640b163f-ede0-49ce-91ed-b6390aac5846)

**Comment by LawJolla** [comment]
After many hours of banging my head on the wall, I have a reproduction.  It's something in my prisma schema.

https://github.com/LawJolla/pothos-prisma-alias-bug

My Prisma schema is one of the oldest around -- first developed when it was GraphCool 2.0 -- so I'm sure it has a lot of cruft.  That said, it's still validating and building on Prisma 5.2.

I tried a few schema changes to isolate it the issue, but no luck so far.  I'll advise if I find something.

Quick video of my change and the result.  

https://github.com/hayes/pothos/assets/17485128/9887ad69-8d9c-4d43-8186-812a6395393c

**Comment by hayes** [comment]
nice, looks like you are getting pretty close to figuring this out.  I'll really interested to see what the actual issue is

**Comment by LawJolla** [comment]
Minimal reproduction is up.

Back in 2018 before I knew any better, I called one of my models "Date"

```
model Date {
...
}
```

is choking the Pothos plugin build.

**Comment by LawJolla** [comment]
Issue figured out, but I'm not sure how to fix it.  (Minimal reproduction here: https://github.com/LawJolla/pothos-prisma-alias-bug )

Back in 2018 before I knew any better, I made a Prisma model named Date.

In Prisma 4.10, my build / Pothos Prisma plugin was fine.  But now with 5.2, the Date model is corrupting the Pothos Prisma Plugin.

In the reproduction, I renamed the Date scalar in case that was the issue.  It doesn't appear to be.

Any thoughts?
